lab_1_classify_by_unigrams/main.py

```diff
@@ -4,51 +4,123 @@
+        else:
+            dictionary[i] = 1/length
+    return dictionary
+
+
+
+
 
 
 def create_language_profile(language: str, text: str) -> dict[str, str | dict[str, float]] | None:
     """
-    Creates a language profile
-    :param language: a language
-    :param text: a text
-    :return: a dictionary with two keys – name, freq
-    """
-
+      Creates a language profile
+      :param language: a language
+      :param text: a text
+      :return: a dictionary with two keys – name, freq
+      """
+    if type(language) != str or type(text) != str:
+        return None
+    new_dictionary = {}
+    text = tokenize(text)
+    freq = calculate_frequencies(text)
+    new_dictionary[language] = freq
+    return new_dictionary
 
 def calculate_mse(predicted: list, actual: list) -> float | None:
     """
-    Calculates mean squared error between predicted and actual values
-    :param predicted: a list of predicted values
-    :param actual: a list of actual values
-    :return: the score
-    """
+       Calculates mean squared error between predicted and actual values
+       :param predicted: a list of predicted values
+       :param actual: a list of actual values
+       :return: the score
+       """
+    if type(predicted) != list or type(actual) != list or len(predicted) != len(actual):
+        return None
+    cubs = 0
+    n = len(actual)
+    for i in range(len(actual)):
+        cubs += (actual[i] - predicted[i]) ** 2
+    mse = cubs / n
+    return mse
+
 
 
 def compare_profiles(
         unknown_profile: dict[str, str | dict[str, float]],
         profile_to_compare: dict[str, str | dict[str, float]]
 ) -> float | None:
     """
-    Compares profiles and calculates the distance using symbols
-    :param unknown_profile: a dictionary of an unknown profile
-    :param profile_to_compare: a dictionary of a profile to compare the unknown profile to
-    :return: the distance between the profiles
-    """
+       Compares profiles and calculates the distance using symbols
+       :param unknown_profile: a dictionary of an unknown profile
+       :param profile_to_compare: a dictionary of a profile to compare the unknown profile to
+       :return: the distance between the profiles
+       """
+    if type(unknown_profile) != dict or type(profile_to_compare) != dict:
+        return None
+    key1 = 'name'
+    key2 = 'freq'
+    if key1 not in unknown_profile or key2 not in unknown_profile or key1 not in profile_to_compare or key2 not in profile_to_compare:
+        return None
+    new_list = []
+    list1 = []
+    list2 = []
+    unknown_freq = unknown_profile['freq']
+    compare_freq = profile_to_compare['freq']
+    unknown_list = list(unknown_freq.values())
+    compare_list = list(compare_freq.values())
+    new_list = unknown_list + compare_list
+    new = list(set(new_list))
+    for i in range(len(new)):
+        if new[i] in unknown_freq.keys():
+            list1.append(unknown_freq[new[i]])
```

Comment:
Consider using dictionary method `get` instead of conditions.

Reply:
Done